src/stores/boardsNew.ts

```diff
@@ -25,68 +48,96 @@ export const useBoardsNewStore = defineStore('boardsNew', () => {
     }, 2000)
   }
 
-  const getBoardsData = async () => {
-    if (userStore.activeUser == null) {
-      await userStore.getUser()
-    }
+  const getBoardsData = (userID: Board['userID']) => {
+    if (userID) {
+      userBoardsColRef.value = query(
+        boardsColRef,
+        where('userID', '==', userID),
+        orderBy('createdAt', 'desc')
+      )
 
-    if (!userStore.activeUser) {
-      throw Error('Failed to retrive user')
-    }
+      onSnapshot(userBoardsColRef.value, (snapshot) => {
+        boards.value = snapshot.docs.map((doc) => {
+          return {
+            docID: doc.id,
+            ...(doc.data() as Omit<Board, 'docID'>)
+          }
+        })
 
-    const requiredDocRef = query(
-      colRef,
-      where('userID', '==', userStore.activeUser.userID)
-    )
-    const docSnap = await getDocs(requiredDocRef)
-    const [documentRef] = docSnap.docs
-    const docRef = doc(db, 'users', documentRef.id)
+        const savedBoard = JSON.parse(
+          localStorage.getItem('currentBoard') || '{}'
+        )
```

Comment:
I wonder, do you actually need localStorage at all?

Reply:
Yes, because as a user you don't have to log out and you'll be automatically logged in the next time you come back to this app. And since you'll be logged in, I should display a proper board as your currently active one. By "proper" I obviously mean - the same you had last time before closing the app.